examples/app_adat_loopback/src/main.xc

```diff
@@ -1,229 +1,148 @@
 // Copyright 2011-2024 XMOS LIMITED.
 // This Software is subject to the terms of the XMOS Public Licence: Version 1.
-// Example code for ADAT. Note that the adat_tx code can be changed to drive a port
-// directly
-
+#include <platform.h>
 #include <xs1.h>
-#include <xclib.h>
 #include "adat_tx.h"
 #include "adat_rx.h"
 #include "stdio.h"
 #include "assert.h"
 
-#define ADAT_TX_DIRECT_PORT_OUT
-
-// Simulator mode:
-// Note: The audio clock is generated by mck_gen(). This has too much jitter for audio but is good enough for loopback testing
-// Loopback of XS1_PORT_1A->XS1_PORT_1B (data) and XS1_PORT_1C->XS1_PORT_1B (clock)
-// must to be activated to make this mode work
-#define XSIM
-
-// Activate trace of received data on 32 bit port
-#define TRACE
-
 #define MAX_GEN_VAL (1<<24)
 #define COUNT_SHIFT 5
 #define DONE_CONDITION_MASK ~((1<<COUNT_SHIFT)-1)
 
-buffered out port:32 p_adat_tx = XS1_PORT_1A;
-buffered in port:32 p_adat_rx = XS1_PORT_1B;
+buffered out port:32 p_adat_tx = PORT_ADAT_OUT;
+buffered in port:32 p_adat_rx = PORT_ADAT_IN;
+in port p_mclk_in = PORT_MCLK_IN;
+out port p_ctrl = PORT_CTRL;
+on tile[1]: clock clk_audio = XS1_CLKBLK_2;
+
+// Found solution: IN 24.000MHz, OUT 24.576000MHz, VCO 2457.60MHz, RD 1, FD 102.400 (m = 2, n = 5), OD 5, FOD 5, ERR 0.0ppm
+// Measure: 100Hz-40kHz: ~8ps
+// 100Hz-1MHz: 63ps.
+// 100Hz high pass: 127ps.
+#define APP_PLL_CTL_24M  0x0A006500
+#define APP_PLL_DIV_24M  0x80000004
+#define APP_PLL_FRAC_24M 0x80000104
+
+// Set secondary (App) PLL control register
+void set_app_pll_init (tileref tile, int app_pll_ctl)
```

Comment:
This should come from lib_sw_pll now ideally.

Reply:
Yes, good point - I just copied this from the examples in lib_spdif. I'll add lib_sw_pll.

Reply:
Ah, we should add an issue to lib_spdif to get those moved too then!